extension/src/experiments/model/index.ts

```diff
@@ -437,28 +438,21 @@ export class ExperimentsModel extends ModelWithPersistence {
     }
   }
 
-  public getRowData() {
+  public getRowData(): Commit[] {
     const commitsBySha = this.applyFiltersToCommits()
 
-    const rows: Commit[] = [
-      { branch: WORKSPACE_BRANCH, ...this.addDetails(this.workspace) }
-    ]
-    for (const { branch, sha } of this.rowOrder) {
-      const commit = commitsBySha[sha]
-      if (!commit) {
-        continue
-      }
-      if (commit.subRows) {
-        commit.subRows = commit.subRows.map(experiment => ({
-          ...experiment,
-          branch
-        }))
-      }
-
-      rows.push({ ...commit, branch })
+    const workspaceBranch = {
+      branch: WORKSPACE_BRANCH,
+      ...this.addDetails(this.workspace)
     }
+    const rowsAcc: Commit[] = []
+    const sorts = this.getSorts()
+    const flattenRows = sorts.length > 0
 
-    return rows
+    for (const { branch, sha } of this.rowOrder) {
```

Comment:
[I] If we have sorts. Let's not collect the data with `subrows` let's have a separate branch of logic which collects the data and returns it. If we don't have sorts leave the logic as it was previously.

(As discussed on a call) This gives us:

1. Breadcrumbs for the table being flat when sorts are applied (think method names)
2. Opportunity to filter correctly when the table is flat (simplify logic for filtering as don't have to retain "filtered parents" if children as unfiltered.
3. No need to collect subrows then mutate the data further (simpler logic overall).

Reply:
Moved the logic to `getFlattenedRowData` that gets called inside of `getRowData` if there are sorts.